### PR TITLE
Add Management API

### PR DESCRIPTION
**Pull request by SanojPunchihewa**
## Purpose
Command Line tool Backend API for managing the WSO2 Micro Integrator

### CODE REVIEW

File: components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/ApiResource.java [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+    private static final String TEMPLATE_ELEMENT = "<Template></Template>";
+    private static final String MAPPING_ELEMENT = "<Mapping></Mapping>";
+    private static final String INSEQ_ELEMENT = "<Inseq></Inseq>";
+    private static final String OUTSEQ_ELEMENT = "<Outseq></Outseq>";
+    private static final String FAULT_ELEMENT = "<Faultseq></Faultseq>";
+
+
+    public ApiResource(String urlTemplate){
+        super(urlTemplate);
+        log.info("Created");
```
Comment (nandika) [inline components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/ApiResource.java]: Let's remove the info logs

File: cmd/build.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
```
Comment (a5anka) [inline cmd/build.sh]: Shall we add the License?

File: cmd/cmd/apiDelete.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+	},
+}
+
+func init() {
+	deleteCmd.AddCommand(apiDeleteCmd)
+
+	// Here you will define your flags and configuration settings.
+
+	apiDeleteCmd.Flags().StringVarP(&apiToDelete, "name", "n", "", "Name of the API to be deleted")
+    apiDeleteCmd.MarkFlagRequired("name")
```
Comment (a5anka) [inline cmd/cmd/apiDelete.go]: Shall we fix inconsistent indentation?

File: cmd/cmd/apiDelete.go [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    apiDeleteCmd.MarkFlagRequired("name")	
+}
+
+func executeDeleteAPICmd(apiname string) {
+
+    err := DeleteAPI(apiname)
+
+    // Result after deleting the API
+    if err == nil {
+
```
Comment (a5anka) [inline cmd/cmd/apiDelete.go]: Do we need extra lines here?

File: cmd/cmd/apiDelete.go [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    //          if count > 0 {
+    //              printAPIs(apis)
+    //          }
+    //      } else {
+    //          utils.Logln(utils.LogPrefixError+"Getting List of APIs", err)
+    //      }
+    //  } else {
+    //      utils.HandleErrorAndExit("Error calling '"+listCmdLiteral+" "+apisCmdLiteral+"'", preCommandErr)
+    //  }
+    // }
```
Comment (a5anka) [inline cmd/cmd/apiDelete.go]: Why do we have commented out code here?

File: cmd/cmd/apiDelete.go [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+// @param name of the api
+// @return error
+func DeleteAPI(name string) (error) {
+
+    finalUrl := utils.RESTAPIBase + utils.PrefixAPIs + "/" + name
+
+    utils.Logln(utils.LogPrefixInfo+"URL:", finalUrl)
+
+    headers := make(map[string]string)
+    // headers[utils.HeaderAuthorization] = utils.HeaderValueAuthPrefixBearer + " " + accessToken
```
Comment (a5anka) [inline cmd/cmd/apiDelete.go]: Why do we have commented out code here?

File: cmd/cmd/apiDelete.go [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+
+    utils.Logln(utils.LogPrefixInfo+"Response:", resp.Status())
+
+    if resp.StatusCode() == http.StatusOK {
+        // apiCarbonAppResponse := &utils.CarbonApp{}
+        // unmarshalError := json.Unmarshal([]byte(resp.Body()), &apiCarbonAppResponse)
+
+        // if unmarshalError != nil {
+        //     utils.HandleErrorAndExit(utils.LogPrefixError+"invalid JSON response", unmarshalError)
+        // }
```
Comment (a5anka) [inline cmd/cmd/apiDelete.go]: Why do we have commented out code here?

Why do we have commented out code here?

File: cmd/cmd/apiInfo.go [hunk 8]
```diff
@@ -0,0 +1,186 @@
+/*
+*  Copyright (c) WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (a5anka) [inline cmd/cmd/apiInfo.go]: Please check whether the license template is correct.

File: cmd/cmd/apiInfo.go [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+    //          if count > 0 {
+    //              printAPIs(apis)
+    //          }
+    //      } else {
+    //          utils.Logln(utils.LogPrefixError+"Getting List of APIs", err)
+    //      }
+    //  } else {
+    //      utils.HandleErrorAndExit("Error calling '"+listCmdLiteral+" "+apisCmdLiteral+"'", preCommandErr)
+    //  }
+    // }
```
Comment (a5anka) [inline cmd/cmd/apiInfo.go]: Why do we have commented out code here?

File: cmd/cmd/apiInfo.go [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+		data = []string{resourceId, "TEMPLATE", resource.Template}
+		table.Append(data)
+		data = []string{resourceId, "MAPPING", resource.Mapping}
+        table.Append(data)
+
+	}
+
+	table.SetBorders(tablewriter.Border{Left: true, Top: true, Right: true, Bottom: false})
+	table.SetRowLine(true)
+    table.SetAutoMergeCells(true)
```
Comment (a5anka) [inline cmd/cmd/apiInfo.go]: Shall we fix inconsistent indentation?

File: cmd/cmd/apiList.go [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+	// 			if count > 0 {
+	// 				printAPIs(apis)
+	// 			}
+	// 		} else {
+	// 			utils.Logln(utils.LogPrefixError+"Getting List of APIs", err)
+	// 		}
+	// 	} else {
+	// 		utils.HandleErrorAndExit("Error calling '"+listCmdLiteral+" "+apisCmdLiteral+"'", preCommandErr)
+	// 	}
+	// }
```
Comment (a5anka) [inline cmd/cmd/apiList.go]: Why do we have commented out code here?

File: components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/Utils.java [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+    public static List<NameValuePair> getQueryParameters(org.apache.axis2.context.MessageContext axis2MessageContext){
+
+        List<NameValuePair> queryParameter = null;
+
+        // extract the query parameters from the Url
+        try {
+            queryParameter = URLEncodedUtils.parse(new URI((String) axis2MessageContext.getProperty(
+                    Constants.Configuration.TRANSPORT_IN_URL)), "UTF-8");
+        }catch (URISyntaxException e){
+            e.printStackTrace();
```
Comment (a5anka) [inline components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/Utils.java]: Shall we not use e.printStackTrace() in production code?

**Comment by a5anka** [comment]
Where can I review the design of this work?

**Comment by a5anka** [comment]
Is there a document describing the high-level design of this work?

**Pull request merged**